### PR TITLE
[FE] Create Add Lesson Modal

### PR DESCRIPTION
**Pull request by b-l-i-n-d**

### CODE REVIEW

File: assets/react/v3/shared/components/modals/AddLessonModal.tsx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        <div css={styles.wrapper}>
+          <div css={styles.lessonInfo}>
+            <Controller
+              name='lesson_name'
+              control={form.control}
+              render={(controllerProps) => (
+                <FormInput
+                  {...controllerProps}
+                  label={__('Lesson Name', 'tutor')}
+                  placeholder={__('Enter Lesson name', 'tutor')}
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/AddLessonModal.tsx:67]:
```suggestion
                  placeholder={__('Enter Lesson Name', 'tutor')}
```

File: assets/react/v3/shared/components/modals/AddLessonModal.tsx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+              )}
+            />
+            <Controller
+              name='description'
+              control={form.control}
+              render={(controllerProps) => (
+                <FormTextareaInput
+                  {...controllerProps}
+                  label={__('Description', 'tutor')}
+                  placeholder={__('Enter Lesson name', 'tutor')}
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/AddLessonModal.tsx:80]:
```suggestion
                  placeholder={__('Enter Lesson Description', 'tutor')}
```

File: assets/react/v3/shared/components/modals/AddLessonModal.tsx [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    width: 1035px;
+    margin: 0 auto;
+    display: grid;
+    grid-template-columns: 1fr 395px;
+    height: 100%;
+  `,
+  lessonInfo: css`
+    padding-block: ${spacing[24]};
+    padding-left: ${spacing[24]};
+    padding-right: ${spacing[64]};
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/AddLessonModal.tsx:200]:
```suggestion
    padding-inline: ${spacing[24]} ${spacing[64]};
```

File: assets/react/v3/shared/components/modals/AddLessonModal.tsx [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    display: flex;
+    flex-direction: column;
+    row-gap: ${spacing[24]};
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/AddLessonModal.tsx:203]:
```suggestion
    gap: ${spacing[24]};
```

File: assets/react/v3/shared/components/modals/AddLessonModal.tsx [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    align-self: start;
+    position: sticky;
+    top: 0;
+  `,
+  rightPanel: css`
+    border-left: 1px solid ${colorTokens.stroke.divider};
+    display: flex;
+    flex-direction: column;
+    row-gap: ${spacing[16]};
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/AddLessonModal.tsx:212]: Do we really need `row-gap`?
```suggestion
    gap: ${spacing[16]};
```

File: assets/react/v3/shared/components/modals/AddLessonModal.tsx [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+    padding-block: ${spacing[24]};
+    padding-right: ${spacing[24]};
+    padding-left: ${spacing[64]};
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/AddLessonModal.tsx:215]:
```suggestion
    padding-inline: ${spacing[64]} ${spacing[24]};
```

File: assets/react/v3/shared/components/modals/AddLessonModal.tsx [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+  `,
+  duration: css`
+    display: flex;
+    align-items: flex-end;
+    column-gap: ${spacing[8]};
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/AddLessonModal.tsx:220]:
```suggestion
    gap: ${spacing[8]};
```

File: assets/react/v3/shared/components/modals/AddLessonModal.tsx [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+  `,
+  durationContent: css`
+    ${typography.small()};
+    color: ${colorTokens.text.hints};
+  `,
+
+  uploadAttachment: css`
+    display: flex;
+    flex-direction: column;
+    row-gap: ${spacing[8]};
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/AddLessonModal.tsx:230]:
```suggestion
    gap: ${spacing[8]};
```

File: assets/react/v3/shared/components/modals/ModalWrapper.tsx [hunk 9]
```diff
@@ -4,14 +4,20 @@ import { borderRadius, Breakpoint, colorTokens, shadow, spacing } from '@Config/
 import { typography } from '@Config/typography';
 import { css } from '@emotion/react';
 import { styleUtils } from '@Utils/style-utils';
+import { MODAL } from '@Config/constants';
+import Show from '@Controls/Show';
 
 interface ModalWrapperProps {
   children: React.ReactNode;
-  onClose: () => void;
+  onClose?: () => void;
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/ModalWrapper.tsx:12]: Why do you make this optional?

File: assets/react/v3/shared/components/modals/ModalWrapper.tsx [hunk 10]
```diff
@@ -37,25 +67,63 @@ export default ModalWrapper;
...
     }
   `,
   header: css`
-    display: flex;
+    display: inline-flex;
     justify-content: space-between;
     align-items: center;
     padding: ${spacing[20]};
     width: 100%;
+    max-height: 72px;
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/ModalWrapper.tsx:91]: Use `height` instead.

File: assets/react/v3/shared/components/modals/ModalWrapper.tsx [hunk 11]
```diff
@@ -37,25 +67,63 @@ export default ModalWrapper;
...
+  `,
+  headerContent: css`
+    display: inline-flex;
+    align-items: center;
+    gap: ${spacing[12]};
+
+    & span {
+      ::before {
+        content: '';
+        border: 0.5px solid ${colorTokens.icon.hints};
```
Comment (ahamed) [inline assets/react/v3/shared/components/modals/ModalWrapper.tsx:104]: Fraction px value may not work all the browsers well. Use nearest integer value.

File: assets/react/v3/shared/config/icon-list.ts [hunk 12]
```diff
@@ -319,6 +319,14 @@ const collection = {
...
+    viewBox: '0 0 24 24',
```
Comment (ahamed) [inline assets/react/v3/shared/config/icon-list.ts:328]: Better use `fill="currentColor"` for modifying the SVG colors.

**Review by ahamed — CHANGES_REQUESTED** [review]

File: assets/react/v3/shared/components/modals/ModalWrapper.tsx (hunk 9 above)
Comment (b-l-i-n-d) [reply assets/react/v3/shared/components/modals/ModalWrapper.tsx:12]: Bhai this was a mistake.

**Review by ahamed — APPROVED** [review]

**Pull request merged**